### ISSUE TITLE
Broken UTC timestamp writing

### ISSUE THREAD

**Issue by WillAyd** [innobi/pantab#190]
Looks like some changes to pandas internals broke the UTC writer. Specifically the `convert_datetime_to_datetimestruct` call in writers.c yields dates close to the unix epoch when they should be in the 2020s

**Comment by WillAyd** [comment]
My guess is that `df._mgr.column_arrays` may have changed between pandas 1.4 and 1.5, the former returning a `datetime64[ns]` and the latter now returning object.

writers.c iterates over the values returned by that call

1.4 behavior:

```python
>>> import pandas as pd
>>> df = pd.DataFrame({"a": [pd.Timestamp("today", tz="utc")]})
>>> df._mgr.column_arrays
[array(['2023-08-30T13:00:06.259921000'], dtype='datetime64[ns]')]
```

1.5
```python
>>> import pandas as pd
>>> df = pd.DataFrame({"a": [pd.Timestamp("today", tz="utc")]})
>>> df._mgr.column_arrays
[array([Timestamp('2023-08-30 13:01:26.997941+0000', tz='UTC')],
      dtype=object)]
```

**Comment by WillAyd** [comment]
Fixed by #187
